src/dmd/typesem.d

```diff
@@ -248,7 +272,7 @@ private void resolveHelper(TypeQualified mt, const ref Loc loc, Scope* sc, Dsymb
                     // https://issues.dlang.org/show_bug.cgi?id=13087
                     // this.field is not constant always
                     if (!v.isThisDeclaration())
-                        goto L3;
+                        return helper3();
```

Comment:
Why is this returning a `void` function?  Is this just another way of writing
```D
helper3();
return;
```

Reply:
Yeah this is an idiom inherited from C++.

Reply:
```
return helper3();
```
is more succint than:
```
{
    helper3();
    return;
}
```